src/Components/GroupForm/GroupForm.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import Button from "../Button/Button.jsx";
+import Select from "../Select/Select.jsx";
+import TextInput from "../TextInput/TextInput.jsx";
+
+import { getPackages, createGroup } from "../../utils/api.js";
+import { languages } from "../../utils/constants.js";
+
+import "./GroupForm.scss";
+
+const GroupForm = ({
+  fixedPackage = false,
+  selectedPackage = null,
+  onSubmitCallback = null,
+}) => {
+  const { t } = useTranslation();
+
+  const [name, setName] = useState("");
+  const [languagePackage, setLanguagePackage] = useState(selectedPackage);
+  const [canSubmit, setCanSubmit] = useState(false);
+  const [languagePackages, setLanguagePackages] = useState([]);
+  const [packageItems, setPackageItems] = useState([]);
+
+  const fetchPackages = useCallback(() => {
+    getPackages(true)
+      .then(({ data }) => {
+        setLanguagePackages(data);
+      })
+      .catch(function (err) {});
+  }, []);
+
+  //make api call to add vocab package
+  async function submitHandler() {
+    const newGroup = {
+      name,
+      active: true,
+    };
+
+    createGroup(languagePackage.id || languagePackage.value, newGroup)
+      .then(({ data }) => {
+        onSubmitCallback && onSubmitCallback(data);
+      })
+      .catch(function (error) {
+        if (error.response.status === 401) {
+          console.log("jwt expired");
+        }
+      });
+  }
+
+  useEffect(() => {
+    setCanSubmit(!(!name || !languagePackage));
+  }, [languagePackage, name]);
+
+  useEffect(() => {
+    if (fixedPackage) {
+      return;
+    }
+
+    fetchPackages();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    setPackageItems(() =>
+      languagePackages.map((p) => {
+        const langs = p.name.split(" - ").map((e) => {
+          const icon = languages.find((x) => x.name === e);
+          return icon ? icon.icon + e : " " + e;
+        });
+
+        return {
+          value: p.id,
+          label: langs.join(" - "),
+        };
+      })
+    );
+  }, [languagePackages]);
+
+  return (
+    <div className="group-form">
+      <div className="form-wrapper">
+        <div className="dropdown">
+          <div className="select-wrapper">
+            <Select
```

Comment:
can we use the custom select option also here?

Reply:
I think we do use the custom select here. Do I miss something?

Reply:
I mean the select with the flags on the right side.